backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -287,52 +289,59 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
 
   /** The instantiated command. */
   lazy val instantiatedCommand: InstantiatedCommand = {
+
     val runtimeEnvironment = RuntimeEnvironmentBuilder(jobDescriptor.runtimeAttributes, jobPaths)(standardParams.minimumRuntimeSettings)
 
-    def adHocFileLocalization(womFile: WomFile): String = womFile.value.substring(womFile.value.lastIndexOf("/") + 1, womFile.value.length)
+    val instantiatedCommandValidation = Command.instantiate(
+      jobDescriptor,
+      backendEngineFunctions,
+      commandLinePreProcessor,
+      commandLineValueMapper,
+      runtimeEnvironment
+    )
 
-    val commandInstantiationInputs = jobDescriptor.evaluatedTaskInputs.map { case (k, v) => k.localName.value -> v }
+    def adHocFileLocalization(womFile: WomFile): String = womFile.value.substring(womFile.value.lastIndexOf("/") + 1)
 
     def validateAdHocFile(value: WomValue): ErrorOr[List[WomFile]] = value match {
-        case f: WomFile => List(f).valid
-        case a: WomArray => a.value.toList.traverse(validateAdHocFile).map(_.flatten)
-        case other => s"Ad-hoc file creation expression invalidly created a ${other.womType.toDisplayString} result.".invalidNel
+      case f: WomFile => List(f).valid
+      case a: WomArray => a.value.toList.traverse(validateAdHocFile).map(_.flatten)
+      case other => s"Ad-hoc file creation expression invalidly created a ${other.womType.toDisplayString} result.".invalidNel
     }
 
     val callable = jobDescriptor.taskCall.callable
+    def makeStringKeyedMap(list: List[(LocalName, WomValue)]): Map[String, WomValue] = list.toMap map { case (k, v) => k.value -> v }
 
-    val adHocFileCreations: ErrorOr[List[WomFile]] = callable.adHocFileCreation.toList.flatTraverse(
-      _.evaluateValue(commandInstantiationInputs, backendEngineFunctions).flatMap(validateAdHocFile)
-    )
+    val command = instantiatedCommandValidation flatMap { instantiatedCommand =>
+      val taskInputs = instantiatedCommand.taskInputs |> makeStringKeyedMap
```

Comment:
Don't you have access to the evaluated task inputs through the jobDescriptor already ? `jobDescritpor.evaluatedTaskInputs`

Reply:
Evaluated inputs yes, preprocessed evaluated inputs no. I'll clean up this naming fail.